Sources/Sentry/Profiling/SentryLaunchProfiling.m

```diff
@@ -135,28 +135,37 @@
 
         SENTRY_LOG_INFO(@"Starting app launch profile at %llu", appLaunchSystemTime);
 
-        // don't worry about synchronizing the write here, as there should be no other tracing
-        // activity going on this early in the process. this codepath is also behind a dispatch_once
-        isTracingAppLaunch = [SentryProfiler startWithTracer:appLaunchTraceId];
+        SentryTransactionContext *context =
+            [[SentryTransactionContext alloc] initWithName:@"launch" operation:@"app.lifecycle"];
+        SentryTracerConfiguration *config = [SentryTracerConfiguration defaultConfiguration];
+        NSDictionary<NSString *, NSNumber *> *rates = appLaunchProfileConfiguration();
+        NSNumber *profilesRate = rates[kSentryLaunchProfileConfigKeyProfilesSampleRate];
+        NSNumber *tracesRate = rates[kSentryLaunchProfileConfigKeyTracesSampleRate];
+        if (profilesRate != nil && tracesRate != nil) {
+            config.profilesSamplerDecision =
+                [[SentrySamplerDecision alloc] initWithDecision:kSentrySampleDecisionYes
+                                                  forSampleRate:profilesRate];
+            context.sampleRate = tracesRate;
+        }
+        launchTracer = [[SentryTracer alloc] initWithTransactionContext:context
+                                                                    hub:nil
+                                                          configuration:config];
     });
 }
 
-BOOL
-injectLaunchSamplerDecisions(
-    SentryTransactionContext *transactionContext, SentryTracerConfiguration *configuration)
+void
+stopLaunchProfile(void)
 {
     NSDictionary<NSString *, NSNumber *> *rates = appLaunchProfileConfiguration();
     removeAppLaunchProfilingConfigBackupFile();
     NSNumber *profilesRate = rates[kSentryLaunchProfileConfigKeyProfilesSampleRate];
     NSNumber *tracesRate = rates[kSentryLaunchProfileConfigKeyTracesSampleRate];
-    if (profilesRate == nil || tracesRate == nil) {
-        return NO;
+    if (launchTracer == nil) {
+        SENTRY_LOG_DEBUG(@"No launch tracer present to stop.");
     }
-    configuration.profilesSamplerDecision =
-        [[SentrySamplerDecision alloc] initWithDecision:kSentrySampleDecisionYes
-                                          forSampleRate:profilesRate];
-    transactionContext.sampleRate = rates[kSentryLaunchProfileConfigKeyTracesSampleRate];
-    return YES;
+
+    SENTRY_LOG_DEBUG(@"Finishing launch tracer.");
```

Comment:
`m`: This code will also run if the `launchTracer` is `nil`. Please add a return statement above or put this into an else statement.

Reply:
Wound up putting a check like that in a subsequent PR